airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -129,10 +136,25 @@ public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws Retr
               StandardSyncSummary standardSyncSummary = standardSyncOutput.get().getStandardSyncSummary();
 
               if (standardSyncSummary != null && standardSyncSummary.getStatus() == ReplicationStatus.FAILED) {
+                failures.addAll(standardSyncOutput.get().getFailures());
+                partialSuccess = standardSyncOutput.get().getStandardSyncSummary().getTotalStats().getRecordsCommitted() > 0;
                 workflowState.setFailed(true);
               }
             } catch (final ChildWorkflowFailure childWorkflowFailure) {
-              if (!(childWorkflowFailure.getCause() instanceof CanceledFailure)) {
+              if (childWorkflowFailure.getCause() instanceof CanceledFailure) {
+                // do nothing, cancellation handled by cancellationScope
+
+              } else if (childWorkflowFailure.getCause() instanceof ActivityFailure) {
```

Comment:
Nit: it can be refactor in:
else {
    switch (chilldWorkflowFailure) {
      case ActivityFailure a: ...
      default: ...     
}
   hrow childWorkflowFailure;
}

From: https://docs.oracle.com/en/java/javase/17/language/pattern-matching-switch-expressions-and-statements.html